File: backend/src/main/java/com/pawconnect/backend/dog/model/Dog.java
package com.pawconnect.backend.dog.model;
import com.pawconnect.backend.user.model.User;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.*;

import java.time.LocalDate;
import java.util.List;


@Entity
@Table(name = "dogs")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Dog {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotBlank
    private String name;

    @NotBlank
    private String breed;

    @NotNull
    private LocalDate birthdate;

    @Enumerated(EnumType.STRING)
    @NotNull
    private DogSize size;

    @Enumerated(EnumType.STRING)
    @NotNull
    private DogGender gender;

    @Enumerated(EnumType.STRING)
    @NotNull
    private Personality personality;

    @Enumerated(EnumType.STRING)
    @NotNull
    private ActivityLevel activityLevel;

    private String about;

    @ElementCollection
    private List<String> photoUrls;

    @ManyToOne
    @JoinColumn(name = "owner_id", nullable = false)
    private User owner;

}

File: mobile/lib/src/models/dog.dart
class Dog {
  final int id;
  final String name;
  final String? breed;
  final String? birthdate;
  final String? size;
  final String? gender;
  final String? personality;
  final String? activityLevel;

  Dog({
    required this.id,
    required this.name,
    this.breed,
    this.birthdate,
    this.size,
    this.gender,
    this.personality,
    this.activityLevel,
  });

  factory Dog.fromJson(Map<String, dynamic> json) {
    return Dog(
      id: json['id'],
      name: json['name'] ?? 'Unnamed',
      breed: json['breed'],
      birthdate: json['birthdate'],
      size: json['size'],
      gender: json['gender'],
      personality: json['personality'],
      activityLevel: json['activityLevel'],
    );
  }
}
